### ISSUE TITLE
clippy::match_ref_pats probably shouldn't fire for if let

### ISSUE THREAD

**Issue by CAD97** [rust-lang/rust-clippy#7740]
Lint name:

I tried this code:

```rust
if let &KeyboardInput {
    keycode: Some(keycode),
    state,
    ..
} = event
{
    // ...
}
```

I expected to see this happen: warning free

Instead, this happened: `clippy::match_ref_pats` matched

The point of `clippy::match_ref_pats` is that it's less notation to put a single deref on the matched value than a ref on every pattern, but with `if let` this doesn't apply, as there's only one pattern to begin with.

Either the description/help message should change when only one pattern is present, or `clippy::match_ref_pats` should only be emitted when more than one pattern are present.

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.56.0-nightly (ad02dc46b 2021-08-26)
binary: rustc
commit-hash: ad02dc46badee510bd3a2c093edf80fcaade91b1
commit-date: 2021-08-26
host: x86_64-pc-windows-msvc
release: 1.56.0-nightly
LLVM version: 13.0.0
```

**Comment by camsteffen** [comment]
For that matter, we should probably not lint `match` if there is just one `&`.

**Comment by 1nF0rmed** [comment]
So, would we be suggesting that for a single `&`, the `clippy::match_ref_pats` should be suppressed?

If so, are there any tests around this and what's the part of the code that I should be looking at for this?

**Comment by 1nF0rmed** [comment]
Thank you for the pointers! 
Just wanted to confirm my approach. Given we are passing the patterns here, should we add a check in the implementation https://github.com/rust-lang/rust-clippy/blob/78e73129baf65f4095002422303157d5e64e9479/clippy_lints/src/matches.rs#L1185

where we iterate through the patterns and count the number of occurrences of `PatKind::Ref(..)`. 
If the count, in that case, is 1, then we just `return`

**Comment by 1nF0rmed** [comment]
Hey, thank you so much for the pointers. are we able to add this to hacktoberfest?

**Comment by 1nF0rmed** [comment]
Can I pick up this issue? @camsteffen

**Comment by camsteffen** [comment]
Yes just comment with `@rustbot claim`
